### DISCUSSION TITLE
Error in moose when submission to cluster

### DISCUSSION THREAD

**Discussion by DaijunHU** [idaholab/moose#20859] [Q&A Getting Started] [answered]
Dear all,

I just re-install moose on the cluster of my institution and installed our own App for use. 
The installation works fine and both moose and my app are good with the  ./run_tests.
However, when I submit it to HPC PBS work, there shows error like:
./c_pfor_am-opt: error while loading shared libraries: libsz.so.2: cannot open shared object file: No such file or directory

So I connect with IT and they copied the library libsz.so.2, then new error like: 
./c_pfor_am-opt: error while loading shared libraries: libglpk.so.36: cannot open shared object file: No such file or directory
So they add this lib file again. Then the error back to libsz.so.2 again.
I feel getting stuck at it so I installed  libaec-1.0.4-1.el7.aarch64.rpm but it did not help.
Is it lacking of some file or lib necesasry in the cluster?
The version in cluster is CentOS Linux release 7.9.2009. 

Please let me know if you have any suggestions and thank you very much in advance.

Best regards,
Daijun

**Comment by milljm** [comment, marked as answer]
My advice is to build and run on one of your compute nodes in an interactive fashion (`qsub -I`). It sounds as if the compute nodes do not have the same libraries and headers being made available on the head node.
